### PR TITLE
Switching restler to use RequestHandler

### PR DESCRIPTION
**Pull request by tomasnorre**
This PR will switch the restler extension from dispatcher to RequestHandler

### CODE REVIEW

**Comment by tomasnorre** [comment]
Please don't just merge please talk to me first, need to coordinate with other tasks.

File: Classes/Http/RestRequestHandler.php [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    protected $bootstrap;
+
+    /**
+     * The request handed over
+     * @var \Psr\Http\Message\ServerRequestInterface
+     */
+    protected $request;
+
+    /**
+     * Constructor handing over the bootstrap and the original request
```
Comment (srotsch) [inline Classes/Http/RestRequestHandler.php]: NiItpicking: the only constructor argument is the bootstrap, I don't see any relation to the incoming request here

File: Classes/Http/RestRequestHandler.php [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    public function __construct(Bootstrap $bootstrap)
+    {
+        $this->bootstrap = $bootstrap;
+    }
+
+    /**
+     * Handles a frontend request
+     *
+     * @param \Psr\Http\Message\ServerRequestInterface $request
+     * @return NULL|\Psr\Http\Message\ResponseInterface|void
```
Comment (srotsch) [inline Classes/Http/RestRequestHandler.php]: Are all of these return types defined in the interface? I'm wondering if we can limit the variations a bit :-)

File: Classes/Http/RestRequestHandler.php [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+    /**
+     * Returns the priority - how eager the handler is to actually handle the
+     * request.
+     *
+     * @return int The priority of the request handler.
+     */
+    public function getPriority()
+    {
+        return 60;
```
Comment (srotsch) [inline Classes/Http/RestRequestHandler.php]: You could make the value a class constant and add some explanatory comment, without a deep-dive into the request handler I would not know why 60 is the value you've chosen.

File: Documentation/Pages/InstallationGuide.rst [hunk 4]
```diff
@@ -55,28 +55,18 @@ Because we are using a forked version of Restler (in order to be able to tag a s
...
-
-  # Allow access to example.com/api for normal REST calls
-  RewriteRule ^api/(.*)$ typo3conf/ext/restler/Scripts/restler_dispatch.php [NC,QSA,L]
-
-  # Allow access to example.com/api_explorer for the online documentation of your API.
-  # You may want to restrict access to this URL.
-  RewriteRule ^api_explorer/(.*)$ typo3conf/ext/restler/Scripts/restler_dispatch.php [NC,QSA,L]
-
+    # The api_explorer/.* MUST not hit TYPO3 (index.php) otherwise the requests will fail.
+    # This / (target) is only to prevent this to happened.
```
Comment (srotsch) [inline Documentation/Pages/InstallationGuide.rst]: happen.

File: ext_emconf.php [hunk 5]
```diff
@@ -13,10 +13,10 @@
     'state' => 'stable',
     'uploadfolder' => 0,
     'clearCacheOnLoad' => 0,
-    'version' => '1.7.3',
+    'version' => '2.0.0-dev',
```
Comment (srotsch) [inline ext_emconf.php]: I assume you'll do a release commit later to remove the -dev suffix?

Comment (tomasnorre) [reply ext_emconf.php]: True.. :)

**Comment by tmotyl** [comment]
looks great!
Any hints regarding upgrade steps for projects using Restler?

**Comment by tomasnorre** [comment]
The documentation should be updated. It's only a matter of adjusting the apache/nginx redirect. 
https://docs.typo3.org/typo3cms/extensions/restler/2.0.0/Pages/InstallationGuide.html

**Comment by tomasnorre** [comment]
In case of any issues/problems please contact me @tmotyl or open an issue.

**Pull request merged**